### PR TITLE
Feature/handle view states

### PR DESCRIPTION
**Pull request by MohammadRezaei92**
Handle Loading and Error view state in market list and detail page

### CODE REVIEW

**Comment by Kaaveh** [comment]
Thanks Mohammad for you lovely PR! We review it as soon as possible! ❤️

File: domain/market/src/main/java/ir/kaaveh/domain/use_case/SyncMarketListUseCase.kt [hunk 1]
```diff
@@ -7,6 +7,6 @@ class SyncMarketListUseCase @Inject constructor(
     private val repository: MarketRepository
 ) {
 
-    suspend operator fun invoke(): Boolean = repository.syncMarketList()
```
Comment (alibahaaa) [inline domain/market/src/main/java/ir/kaaveh/domain/use_case/SyncMarketListUseCase.kt:10]: Tests failed, update NewsRepositoryImplTest class please. @MohammadRezaei92

**Review by alibahaaa — CHANGES_REQUESTED** [review]
Thanks @MohammadRezaei92 jan for contribution, While we can merge it now after solving failded test, it might be better to handle "improve data sync flow" task in a separate PR for better organization, anyway, great job on the PR!

**Review by alibahaaa — APPROVED** [review]
Well done @MohammadRezaei92 jan.

**Pull request merged**